v2/pkg/runner/enumerate.go

```diff
@@ -147,3 +147,36 @@ func (r *Runner) EnumerateSingleDomain(ctx context.Context, domain string, outpu
 
 	return nil
 }
+func (r *Runner) findMatchingSubdomain(pattern, value string) (bool, error) {
+	pattern = strings.ReplaceAll(pattern, ".", "\\.")
```

Comment:
Precompile and validate match/filtering regexes/keywords in [validateOptions](https://github.com/projectdiscovery/subfinder/blob/cb6fbbb59f1af05c1b50e2b05320d7540ac3aadc/v2/pkg/runner/validate.go#L12), they can be stored in a private slice field of `Options`, for example:
```go
type Options struct {
...
matchRegexes      []*regexp.Regexp
filterRegexes []*regexp.Regexp
...
}
```

Reply:
Done